db/db.go

```diff
@@ -78,19 +78,31 @@ type DB interface {
 
 	// Gateways returns gateways with the given pagination options.
 	Gateways(offset, limit int) ([]tx.Tx, error)
+
+	// GatewayCount returns the number of gateways persisted
+	GatewayCount() (int, error)
+
+	// GatewayCount returns the number of gateways persisted
+	MaxGatewayCount() int
 }
 
 type database struct {
-	db *sql.DB
+	db              *sql.DB
+	maxGatewayCount int
 }
 
 // New creates a new DB instance.
-func New(db *sql.DB) DB {
+func New(db *sql.DB, maxGatewayCount int) DB {
```

Comment:
I think it would be nicer for this to accept some `Options` but this is fine for now.